frontend/packages/dev-console/src/components/topology/filters/TopologyFilterBar.tsx

```diff
@@ -99,11 +121,12 @@ const mergeProps = (
     onFiltersChange({ ...filters, display });
   },
   onSearchQueryChange: (searchQuery) => {
-    onFiltersChange({ ...filters, searchQuery });
     if (searchQuery.length > 0) {
       setQueryArgument(TOPOLOGY_SEARCH_FILTER_KEY, searchQuery);
+      document.body.classList.add('odc-m-filter-active');
     } else {
       removeQueryArgument(TOPOLOGY_SEARCH_FILTER_KEY);
+      document.body.classList.remove('odc-m-filter-active');
```

Comment:
I'd prefer to see this done in a callback for `onSeachChange` in the `Topology` component. It should probably be done _only_ there and removed from the `GraphComponent`.  That way it is controlled by a single component.

Reply:
You mean other class changes that are there in `GraphComponent` right now should also be moved to `Topology` component?

Reply:
yes, as well as from a callback done here.

Reply:
@jeff-phillips-18 Done. Moved the logic into `onSearchChange` in `Topology`.